src/bits/find_if.h

```diff
@@ -84,17 +85,66 @@ class iterator_result
     Iterator end;
 };
 
+template <typename Iterator>
+class iterator_result_rvalue
+{
+public:
+    using value_type = typename Iterator::value_type;
+    iterator_result_rvalue(Iterator iterator, Iterator end)
+        : m_has_value(iterator != end)
+    {
+        if (m_has_value)
+            m_value = std::make_unique<value_type>(std::move(*iterator));
+    }
+    operator bool() const { return m_has_value; }
+    bool has_result() const { return m_has_value; }
+    decltype(auto) operator*() const
+    {
+        assert(m_has_value);
+        return *m_value;
+    }
+    decltype(auto) operator->() const
+    {
+        assert(m_has_value);
+        return &(*m_value);
+    }
+
+    bool m_has_value;
```

Comment:
This bool is no longer necessary, like std::optional, std::unique_ptr can be "empty" (i.e. nullptr).
So just checking whether `m_value.get()` is nullptr is enough.
e.g. use: https://en.cppreference.com/w/cpp/memory/unique_ptr/operator_bool

Reply:
indeed, thx